R/digest.R

```diff
@@ -107,7 +106,7 @@ digest <- function(object, algo=c("md5", "sha1", "crc32", "sha256", "sha512",
     if (!is_streaming_algo) {
         val <- .Call(digest_impl,
                      object,
-                     as.integer(algoint),
+                     algoint,
                      as.integer(length),
                      as.integer(skip),
                      as.integer(raw),
```

Comment:
The entire block is actually "wrong" in the sense of when I wrote it I had likely just learned about `.Call()` over `.C()` and the latter requires this -- but the former does not.  

So we could extend this PR (if you're game) to avoiding / minimising the casting of the other arguments length, skip and raw.  (I will have to check later what type of variable is used at the C level.)

Reply:
I am thinking of my approach to this as being as each optimization feature comes in, remove the cast here. (definitely these casts can go away, as the casting can be handled correctly on the C side by the SEXP interpreters.

Reply:
which is to say, yes, I'll definitely address these. Question: do you want them addressed in *this* PR or as part of *this series*?

Reply:
We can take it one at a time too.  

Now, out of curiousity, and I buy into the PR just for "cleaner expression of intent" (ie algo choice as int) but what performance gain do you claim this gets, and is that not likely only relevant for very frequent calls with really small payloads?  They exist, but also consider the `fortune` below ...

```r
> fortunes::fortune("save")

Paul Gilbert: [code comparing speed of apply(z,2,sum) vs. rep(1,10000)%*%z)] which seemed completely contrary to all my childhood teachings.
Douglas Bates: Must have had an interesting childhood if you spent it learning about the speeds of various matrix multiplication techniques.
Paul Gilbert: [...] why is apply so slow?
Brian Ripley: 'so slow' sic: what are you going to do in the 7ms you saved?
   -- Paul Gilbert, Douglas Bates, and Brian D. Ripley (discussing 'the incredible lightness of crossprod')
      R-devel (January 2005)

> 
```

Reply:
I have yet to systematically look at any "large" payload cases.  For "small" payload cases, profvis indicates that the match.arg lines are equal to or greater than the actual Call itself.

I have checked 1m character objects, and these parts of the R go down to ~10% of the total (instead of >=2/3rd).